### DISCUSSION TITLE
juicefs 选在tikv 作为元数据存储，之前有考虑过foundationdb?

### DISCUSSION THREAD

**Discussion by melin** [juicedata/juicefs#1396] [Q&A] [answered]
最近也在考虑元数据存储选型，juicefs 选在tikv 作为元数据存储，之前有考虑过foundationdb?   国外的snowflake, firebolt 在使用foundationdb，作为一个从参考。

**Comment by xiaogaozi** [comment, marked as answer]
你好，支持 FoundationDB 已经有一个 PR：https://github.com/juicedata/juicefs/pull/644

**Comment by davies** [comment]
选择 TiKV 的原因是国内的用户对 TiKV 更熟悉，更容易接受它，我们跟 TiKV 团队也更容易一起改进方案。FoundationDB 的社区感觉没那么活跃，我们的 PR 准备有半年了，基本没人有兴趣试用。

**Reply by aep** [reply]
hey @davies , has that changed in the last 2 years? I feel the opposite, tikv is pretty much dead. new features like v2 api dont even work without tidb. foundationdb is still.. not very active but at least it seems to be kept alive?

**Reply by davies** [reply]
No, TiKV is still actively maintained by community.  The V2 API is not used by JuiceFS yet, so JuiceFS does not depend on TiDB.

Currently, There are many big TiKV cluster used for JuiceFS, but no many stories about FoundationDB is used with JuiceFS yet, so we think TiKV is still the better choice for JuiceFS.

**Reply by aep** [reply]
thanks for the response. do you think v2 would be worth it? it does solve the separation issue so you can use one tikv cluster for both meta and data. their k8s operator unfortunately supports neither tikv nor api v2, so not sure if it has any future

**Reply by sunxiaoguang** [reply]
Hello friends,

I believe it is worth mentioning that API V2 does not depend on TiDB. It was actually designed to run multiple applications on top of the same TiKV cluster, with TiDB being one of them. Currently, both client-go and client-rust have built-in support for keyspaces. While there may be some features available in one client but not in the other, it is definitely possible to support all features across all clients.

**Reply by sunxiaoguang** [reply]
Talking about TiDB Operator, It's does not stop you from using API V2. Just enable API V2 with appropriate configuration. And you are able to connect to TiKV with API V2 clients.
